### PR TITLE
Add Transactions View Timeline Chart to Admin Dashboard Reports

### PR DESCRIPTION
**Pull request by SudharakaP**
Related to https://github.com/opencollective/opencollective/issues/4658

Require: https://github.com/opencollective/opencollective-api/pull/6640

⚠️ Should be merged after completion of https://github.com/opencollective/opencollective-frontend/pull/6873

https://user-images.githubusercontent.com/12435965/135578015-1127155e-2dbb-49f2-9f36-a826c6ff9efc.mp4

### CODE REVIEW

File: components/host-dashboard/HostDashboardReports.js [hunk 1]
```diff
@@ -150,7 +170,15 @@ const getDefaultDateInterval = () => {
 const HostDashboardReports = ({ hostSlug }) => {
   const [dateInterval, setDateInterval] = React.useState(getDefaultDateInterval);
   const [collectives, setCollectives] = React.useState(null);
-  const { data, error, loading } = useQuery(hostReportPageQuery, { variables: { hostSlug }, context: API_V2_CONTEXT });
+  const { data, error, loading } = useQuery(hostReportPageQuery, {
+    variables: {
+      hostSlug,
+      dateFrom: new Date(dateInterval?.from),
+      dateTo: new Date(dateInterval?.to),
+      account: collectives,
```
Comment (Betree) [inline components/host-dashboard/HostDashboardReports.js:178]: Oops, looks like I forgot to add these in my previous PR. I'll open one to make sure we fix this ASAP

File: components/host-dashboard/reports-section/HostDownloadsSection.js [hunk 2]
```diff
@@ -50,7 +50,7 @@ const triggerCSVDownload = (host, reportUrl, dateInterval) => {
 };
 
 const HostDownloadsSection = ({ host, collectives, dateInterval }) => {
-  const accountsSlugs = collectives?.map(c => c.value.slug);
+  const accountsSlugs = collectives?.map(c => c.value?.slug);
```
Comment (Betree) [inline components/host-dashboard/reports-section/HostDownloadsSection.js:53]: I fixed this in https://github.com/opencollective/opencollective-frontend/pull/7110

File: components/host-dashboard/reports-section/TransactionsOverviewSection.js [hunk 3]
```diff
@@ -1,13 +1,133 @@
...
+        formatter: function (value) {
+          return value < 1000 ? value : `${Math.round(value / 1000)}k`;
+        },
+      },
+    },
+  };
+};
+
+const getNumberOfDays = (startDate, endDate) => {
+  const startTimeOfStatistics = new Date(2015, 0, 1);
```
Comment (Betree) [inline components/host-dashboard/reports-section/TransactionsOverviewSection.js:53]: Why January 1st 2015? If we need a fallback, it should be `host.createdAt` as we don't want to go back to 2015 for a host created in 2020.

File: components/host-dashboard/reports-section/TransactionsOverviewSection.js [hunk 4]
```diff
@@ -1,13 +1,133 @@
...
+  if (startDate && endDate) {
+    numberOfDays = days(new Date(startDate), new Date(endDate));
+  } else if (startDate) {
+    numberOfDays = days(new Date(startDate));
+  } else if (endDate) {
+    numberOfDays = days(startTimeOfStatistics, new Date(endDate));
+  } else {
+    numberOfDays = days(startTimeOfStatistics);
+  }
+  return numberOfDays;
```
Comment (Betree) [inline components/host-dashboard/reports-section/TransactionsOverviewSection.js:64]: We can probably simplify with something like

```suggestion
  return dayjs(startDate || host.createdAt).diff(dayjs(endDate), 'day');
```

File: components/host-dashboard/reports-section/TransactionsOverviewSection.js [hunk 5]
```diff
@@ -140,20 +260,61 @@ const getTransactionsBreakdownChartData = host => {
...
+        ) : (
+          <div>
+            <ProportionalAreaChart areas={areaChartData} borderRadius="6px 6px 0 0" />
+            <ProportionalAreaChart areas={transactionBreakdownChart} borderRadius="0 0 6px 6px" />
+          </div>
+        )}
+      </Box>
+      <Box>
+        {isLoading ? (
+          <Loading />
```
Comment (Betree) [inline components/host-dashboard/reports-section/TransactionsOverviewSection.js:302]: For consistency, can we use a `LoadingPlaceholder` with the same size as the chart it's replacing?

File: components/host-dashboard/reports-section/TransactionsOverviewSection.js (hunk 3 above)
Comment (SudharakaP) [reply components/host-dashboard/reports-section/TransactionsOverviewSection.js]: I've corrected this to include the `host.createdAt` date. Thanks. 👍🏽

File: components/host-dashboard/reports-section/TransactionsOverviewSection.js (hunk 5 above)
Comment (SudharakaP) [reply components/host-dashboard/reports-section/TransactionsOverviewSection.js]: Done. 👍🏽

File: components/host-dashboard/reports-section/TransactionsOverviewSection.js (hunk 4 above)
Comment (SudharakaP) [reply components/host-dashboard/reports-section/TransactionsOverviewSection.js]: Actually, we already have, `days` function in `util.js` and I think we can get rid of this function and use that instead, which I've done. 😉

File: components/host-dashboard/reports-section/TransactionsOverviewSection.js [hunk 6]
```diff
@@ -1,13 +1,117 @@
...
+const constructDataPointObjects = (category, dataPoints) => {
+  let dataPointObject;
+  if (category === 'YEAR') {
+    dataPointObject = new Array(6).fill(0);
+    const currentYear = new Date().getFullYear();
+    dataPoints.forEach(dataPoint => {
+      const year = new Date(dataPoint.date).getFullYear();
+      if (year > currentYear - 6) {
+        dataPointObject[5 - (currentYear - year)] = dataPoint.amount.value;
+      }
```
Comment (Betree) [inline components/host-dashboard/reports-section/TransactionsOverviewSection.js:91]: This is difficult to parse. Can we add some comments? Also for the other ones (month, week)

File: components/host-dashboard/reports-section/TransactionsOverviewSection.js [hunk 7]
```diff
@@ -1,13 +1,117 @@
...
+        formatter: function (value) {
+          return value < 1000 ? value : `${Math.round(value / 1000)}k`;
+        },
+      },
+    },
+  };
+};
+
+const getCategories = (intl, startDate, endDate, hostCreatedAt) => {
+  const numberOfDays = days(startDate || hostCreatedAt, endDate);
```
Comment (Betree) [inline components/host-dashboard/reports-section/TransactionsOverviewSection.js:52]: :ice_cream:  I don't feel like using our own helper brings any value here, it adds complexity and opens the door for errors while `dayjs` provides something tested and simple:

```suggestion
  const numberOfDays = dayjs(startDate || host.createdAt).diff(dayjs(endDate), 'day');
```

File: components/host-dashboard/reports-section/TransactionsOverviewSection.js (hunk 6 above)
Comment (SudharakaP) [reply components/host-dashboard/reports-section/TransactionsOverviewSection.js:94]: I've added some comments to explain what we are doing. Let me know if that's enough or if we need more clarification. 🤔

**Review by Betree — COMMENTED** [review]
Not a requested change: this `constructDataPointObjects` is a bit complex and looks like it could be refactored in smaller helpers. Let's not iterate on this now, we'll just fix the blockers below so that we can merge asap.

File: components/host-dashboard/reports-section/TransactionsOverviewSection.js [hunk 8]
```diff
@@ -1,13 +1,126 @@
...
+  }
+};
+
+const constructDataPointObjects = (category, dataPoints) => {
+  let dataPointObject;
+  /*
+   * Show data for the past 5 years form the current year.
+   */
+  if (category === 'YEAR') {
+    dataPointObject = new Array(6).fill(0);
```
[the remark this answers is not in the document]
Comment (SudharakaP) [reply components/host-dashboard/reports-section/TransactionsOverviewSection.js:88]: Thanks. I've corrected this one. 👍🏽

**Comment by SudharakaP** [comment]
> Not a requested change: this `constructDataPointObjects` is a bit complex and looks like it could be refactored in smaller helpers. Let's not iterate on this now, we'll just fix the blockers below so that we can merge asap.

That's true. Let me open a new issue for this so that I don't forget; https://github.com/opencollective/opencollective/issues/4958

File: components/host-dashboard/reports-section/TransactionsOverviewSection.js [hunk 9]
```diff
@@ -1,13 +1,121 @@
...
+    },
+  };
+};
+
+const getCategories = (intl, startDate, endDate, hostCreatedAt) => {
+  const numberOfDays = dayjs(startDate || hostCreatedAt).diff(dayjs(endDate), 'day');
+  if (numberOfDays <= 7) {
+    const startDay = startDate.getDay();
+    return [...new Array(7)].map(
+      (_, idx) => `${intl.formatDate(new Date(0, 0, idx + startDay), { weekday: 'long' }).toUpperCase()}`,
```
Comment (Betree) [inline components/host-dashboard/reports-section/TransactionsOverviewSection.js:56]: Why use a template string? Same for the other two below

```suggestion
      (_, idx) => intl.formatDate(new Date(0, 0, idx + startDay), { weekday: 'long' }).toUpperCase(),
```

**Review by Betree — APPROVED** [review]
Let's merge and iterate if needed

File: components/host-dashboard/reports-section/TransactionsOverviewSection.js (hunk 9 above)
Comment (SudharakaP) [reply components/host-dashboard/reports-section/TransactionsOverviewSection.js]: Ah, seems a mistake. 🤯 I've removed these.

**Pull request merged**